simvue/api/objects/grids.py

```diff
@@ -79,11 +79,14 @@ def on_reconnect(self, id_mapping: dict[str, str]) -> None:
         id_mapping : dict[str, str]
             mapping from offline identifier to new online identifier.
         """
-        for run_id, metric_name in self._staging.pop("runs", []):
+        _online_runs = [
+            (id_mapping[run_id], metric_name)
+            for run_id, metric_name in self._staging.pop("runs", [])
+        ]
+        super().commit()  # TODO this means the sender iwll commit twice - that ok?
```

Comment:
A commit should generally wipe anything that has been sent, so doing it twice is not a bad thing, it just becomes meaningless if nothing new has been added